models/classes/class.QtiTestService.php

```diff
@@ -873,19 +875,46 @@ private function deleteTestsFromClassByLabel(
         core_kernel_classes_Class $itemClass
     ): void {
         $testService = $this->getTestService();
-        $itemTreeService = $this->getItemTreeService();
+
+        $deletedTests = [];
+        $deletedItemClasses = [];
+        $refs = [];
 
         foreach ($testClass->getInstances() as $testInstance) {
             if ($testInstance->getLabel() === $testLabel) {
+                $deletedTests[] = $testInstance->getUri();
                 $testService->deleteTest($testInstance);
             }
         }
 
         foreach ($itemClass->getSubClasses() as $subClass) {
             if ($subClass->getLabel() === $itemsClassLabel) {
-                $itemTreeService->deleteClass($subClass);
+                $deletedItemClasses[] = $subClass->getUri();
+                $refs = array_merge($refs, $this->deleteItemClass($subClass));
```

Comment:
Maybe we could keep the previous logic here and just add https://github.com/oat-sa/extension-tao-testqti/pull/2393/files#r1260920766